eras/alonzo/impl/testlib/Test/Cardano/Ledger/Alonzo/Imp/UtxowSpec/Invalid.hs

```diff
@@ -2,228 +2,347 @@
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE GADTs #-}
 {-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NumericUnderscores #-}
 {-# LANGUAGE OverloadedLists #-}
-{-# LANGUAGE PatternSynonyms #-}
 {-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TupleSections #-}
 {-# LANGUAGE TypeApplications #-}
 
 module Test.Cardano.Ledger.Alonzo.Imp.UtxowSpec.Invalid (spec) where
 
 import Cardano.Ledger.Address (Addr (..))
+import Cardano.Ledger.Allegra.Scripts (AllegraEraScript (..))
+import Cardano.Ledger.Alonzo (Alonzo)
 import Cardano.Ledger.Alonzo.Core (
+  AlonzoEraScript (..),
   AlonzoEraTxWits (..),
-  networkIdTxBodyL,
+  AsItem (..),
+  AsIx (AsIx),
+  PlutusPurpose,
+  collateralInputsTxBodyL,
+  eraMaxLanguage,
+  mkBasicTx,
+  mkBasicTxBody,
+  mkBasicTxOut,
+  mkCertifyingPurpose,
+  mkDelegStakeTxCert,
+  mkMintingPurpose,
+  mkRegTxCert,
+  mkSpendingPurpose,
+  mkUnRegTxCert,
   scriptIntegrityHashTxBodyL,
+  toSpendingPurpose,
  )
 import Cardano.Ledger.Alonzo.Plutus.Evaluate (CollectError (..))
 import Cardano.Ledger.Alonzo.Rules (
-  AlonzoUtxoPredFailure (..),
-  AlonzoUtxosPredFailure (..),
+  AlonzoUtxosPredFailure (CollectErrors),
   AlonzoUtxowPredFailure (..),
  )
-import Cardano.Ledger.Alonzo.Scripts
+import Cardano.Ledger.Alonzo.Scripts (eraLanguages)
 import Cardano.Ledger.Alonzo.TxOut (dataHashTxOutL)
-import Cardano.Ledger.Alonzo.TxWits (Redeemers (..), TxDats (..))
-import Cardano.Ledger.BaseTypes
+import Cardano.Ledger.Alonzo.TxWits (Redeemers (..), TxDats (..), unRedeemers)
+import Cardano.Ledger.BaseTypes (Network (..), StrictMaybe (..), inject, natVersion)
 import Cardano.Ledger.Coin (Coin (..))
-import Cardano.Ledger.Core
+import Cardano.Ledger.Core (
+  InjectRuleFailure (..),
+  addrTxWitsL,
+  bodyTxL,
+  certsTxBodyL,
+  eraProtVerLow,
+  inputsTxBodyL,
+  outputsTxBodyL,
+  scriptTxWitsL,
+  witsTxL,
+ )
```

Comment:
No explicit imports on Core either. Core module from the era should be used instead
```suggestion
import Cardano.Ledger.Alonzo.Core
```